// @uiw/codemirror-extensions-langs/cjs/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.langs = exports.langNames = void 0;
exports.loadLanguage = loadLanguage;
var _language = require("@codemirror/language");
var _langCpp = require("@codemirror/lang-cpp");
var _langCss = require("@codemirror/lang-css");
var _langGo = require("@codemirror/lang-go");
var _langHtml = require("@codemirror/lang-html");
var _langJava = require("@codemirror/lang-java");
var _langJavascript = require("@codemirror/lang-javascript");
var _langJinja = require("@codemirror/lang-jinja");
var _langJson = require("@codemirror/lang-json");
var _langLess = require("@codemirror/lang-less");
var _langLiquid = require("@codemirror/lang-liquid");
var _langMarkdown = require("@codemirror/lang-markdown");
var _langPhp = require("@codemirror/lang-php");
var _langPython = require("@codemirror/lang-python");
var _langRust = require("@codemirror/lang-rust");
var _langSass = require("@codemirror/lang-sass");
var _langWast = require("@codemirror/lang-wast");
var _langXml = require("@codemirror/lang-xml");
var _langYaml = require("@codemirror/lang-yaml");
var _apl2 = require("@codemirror/legacy-modes/mode/apl");
var _asciiarmor = require("@codemirror/legacy-modes/mode/asciiarmor");
var _asn2 = require("@codemirror/legacy-modes/mode/asn1");
var _brainfuck = require("@codemirror/legacy-modes/mode/brainfuck");
var _cobol = require("@codemirror/legacy-modes/mode/cobol");
var _clike = require("@codemirror/legacy-modes/mode/clike");
var _clojure = require("@codemirror/legacy-modes/mode/clojure");
var _css2 = require("@codemirror/legacy-modes/mode/css");
var _cmake2 = require("@codemirror/legacy-modes/mode/cmake");
var _coffeescript = require("@codemirror/legacy-modes/mode/coffeescript");
var _commonlisp = require("@codemirror/legacy-modes/mode/commonlisp");
var _cypher2 = require("@codemirror/legacy-modes/mode/cypher");
var _python2 = require("@codemirror/legacy-modes/mode/python");
var _crystal = require("@codemirror/legacy-modes/mode/crystal");
var _d2 = require("@codemirror/legacy-modes/mode/d");
var _diff2 = require("@codemirror/legacy-modes/mode/diff");
var _dtd2 = require("@codemirror/legacy-modes/mode/dtd");
var _dylan2 = require("@codemirror/legacy-modes/mode/dylan");
var _ecl2 = require("@codemirror/legacy-modes/mode/ecl");
var _eiffel = require("@codemirror/legacy-modes/mode/eiffel");
var _elm2 = require("@codemirror/legacy-modes/mode/elm");
var _erlang = require("@codemirror/legacy-modes/mode/erlang");
var _factor2 = require("@codemirror/legacy-modes/mode/factor");
var _forth2 = require("@codemirror/legacy-modes/mode/forth");
var _fortran = require("@codemirror/legacy-modes/mode/fortran");
var _mllike = require("@codemirror/legacy-modes/mode/mllike");
var _gas = require("@codemirror/legacy-modes/mode/gas");
var _gherkin = require("@codemirror/legacy-modes/mode/gherkin");
var _groovy2 = require("@codemirror/legacy-modes/mode/groovy");
var _haskell = require("@codemirror/legacy-modes/mode/haskell");
var _haxe = require("@codemirror/legacy-modes/mode/haxe");
var _idl = require("@codemirror/legacy-modes/mode/idl");
var _javascript = require("@codemirror/legacy-modes/mode/javascript");
var _julia = require("@codemirror/legacy-modes/mode/julia");
var _livescript = require("@codemirror/legacy-modes/mode/livescript");
var _lua2 = require("@codemirror/legacy-modes/mode/lua");
var _mirc = require("@codemirror/legacy-modes/mode/mirc");
var _mathematica = require("@codemirror/legacy-modes/mode/mathematica");
var _modelica = require("@codemirror/legacy-modes/mode/modelica");
var _mumps = require("@codemirror/legacy-modes/mode/mumps");
var _mbox2 = require("@codemirror/legacy-modes/mode/mbox");
var _nsis = require("@codemirror/legacy-modes/mode/nsis");
var _ntriples = require("@codemirror/legacy-modes/mode/ntriples");
var _octave = require("@codemirror/legacy-modes/mode/octave");
var _oz2 = require("@codemirror/legacy-modes/mode/oz");
var _pascal = require("@codemirror/legacy-modes/mode/pascal");
var _perl = require("@codemirror/legacy-modes/mode/perl");
var _pig2 = require("@codemirror/legacy-modes/mode/pig");
var _powershell = require("@codemirror/legacy-modes/mode/powershell");
var _properties2 = require("@codemirror/legacy-modes/mode/properties");
var _protobuf = require("@codemirror/legacy-modes/mode/protobuf");
var _pug2 = require("@codemirror/legacy-modes/mode/pug");
var _puppet = require("@codemirror/legacy-modes/mode/puppet");
var _q2 = require("@codemirror/legacy-modes/mode/q");
var _r2 = require("@codemirror/legacy-modes/mode/r");
var _rpm = require("@codemirror/legacy-modes/mode/rpm");
var _ruby = require("@codemirror/legacy-modes/mode/ruby");
var _sas2 = require("@codemirror/legacy-modes/mode/sas");
var _scheme = require("@codemirror/legacy-modes/mode/scheme");
var _shell = require("@codemirror/legacy-modes/mode/shell");
var _sieve2 = require("@codemirror/legacy-modes/mode/sieve");
var _smalltalk = require("@codemirror/legacy-modes/mode/smalltalk");
var _sparql2 = require("@codemirror/legacy-modes/mode/sparql");
var _stylus = require("@codemirror/legacy-modes/mode/stylus");
var _swift2 = require("@codemirror/legacy-modes/mode/swift");
var _stex = require("@codemirror/legacy-modes/mode/stex");
var _verilog = require("@codemirror/legacy-modes/mode/verilog");
var _tcl2 = require("@codemirror/legacy-modes/mode/tcl");
var _textile2 = require("@codemirror/legacy-modes/mode/textile");
var _toml2 = require("@codemirror/legacy-modes/mode/toml");
var _troff = require("@codemirror/legacy-modes/mode/troff");
var _ttcn2 = require("@codemirror/legacy-modes/mode/ttcn");
var _ttcnCfg = require("@codemirror/legacy-modes/mode/ttcn-cfg");
var _turtle = require("@codemirror/legacy-modes/mode/turtle");
var _webidl = require("@codemirror/legacy-modes/mode/webidl");
var _vb2 = require("@codemirror/legacy-modes/mode/vb");
var _vbscript = require("@codemirror/legacy-modes/mode/vbscript");
var _velocity = require("@codemirror/legacy-modes/mode/velocity");
var _vhdl2 = require("@codemirror/legacy-modes/mode/vhdl");
var _xquery = require("@codemirror/legacy-modes/mode/xquery");
var _yacas = require("@codemirror/legacy-modes/mode/yacas");
var _z2 = require("@codemirror/legacy-modes/mode/z80");
var _mscgen2 = require("@codemirror/legacy-modes/mode/mscgen");
var _langVue = require("@codemirror/lang-vue");
var SQL = _interopRequireWildcard(require("@codemirror/lang-sql"));
var _codemirrorLangNix = require("@replit/codemirror-lang-nix");
var _codemirrorLangSvelte = require("@replit/codemirror-lang-svelte");
var _codemirrorLangSolidity = require("@replit/codemirror-lang-solidity");
// auto-generated by gen-langs-map.cjs – DO NOT EDIT

var langs = exports.langs = {
  '1': function _() {
    return _language.StreamLanguage.define(_troff.troff);
  },
  '2': function _() {
    return _language.StreamLanguage.define(_troff.troff);
  },
  '3': function _() {
    return _language.StreamLanguage.define(_troff.troff);
  },
  '4': function _() {
    return _language.StreamLanguage.define(_troff.troff);
  },
  '4th': function th() {
    return _language.StreamLanguage.define(_forth2.forth);
  },
  '5': function _() {
    return _language.StreamLanguage.define(_troff.troff);
  },
  '6': function _() {
    return _language.StreamLanguage.define(_troff.troff);
  },
  '7': function _() {
    return _language.StreamLanguage.define(_troff.troff);
  },
  '8': function _() {
    return _language.StreamLanguage.define(_troff.troff);
  },
  '9': function _() {
    return _language.StreamLanguage.define(_troff.troff);
  },
  apl: function apl() {
    return _language.StreamLanguage.define(_apl2.apl);
  },
  asc: function asc() {
    return _language.StreamLanguage.define(_asciiarmor.asciiArmor);
  },
  asn: function asn() {
    return _language.StreamLanguage.define((0, _asn2.asn1)({}));
  },
  asn1: function asn1() {
    return _language.StreamLanguage.define((0, _asn2.asn1)({}));
  },
  b: function b() {
    return _language.StreamLanguage.define(_brainfuck.brainfuck);
  },
  bash: function bash() {
    return _language.StreamLanguage.define(_shell.shell);
  },
  bf: function bf() {
    return _language.StreamLanguage.define(_brainfuck.brainfuck);
  },
  BUILD: function BUILD() {
    return (0, _langPython.python)();
  },
  bzl: function bzl() {
    return (0, _langPython.python)();
  },
  c: function c() {
    return (0, _langCpp.cpp)();
  },
  'c++': function c() {
    return (0, _langCpp.cpp)();
  },
  cc: function cc() {
    return (0, _langCpp.cpp)();
  },
  cfg: function cfg() {
    return _language.StreamLanguage.define(_ttcnCfg.ttcnCfg);
  },
  cjs: function cjs() {
    return (0, _langJavascript.javascript)();
  },
  cl: function cl() {
    return _language.StreamLanguage.define(_commonlisp.commonLisp);
  },
  clj: function clj() {
    return _language.StreamLanguage.define(_clojure.clojure);
  },
  cljc: function cljc() {
    return _language.StreamLanguage.define(_clojure.clojure);
  },
  cljs: function cljs() {
    return _language.StreamLanguage.define(_clojure.clojure);
  },
  cljx: function cljx() {
    return _language.StreamLanguage.define(_clojure.clojure);
  },
  cmake: function cmake() {
    return _language.StreamLanguage.define(_cmake2.cmake);
  },
  'cmake.in': function cmakeIn() {
    return _language.StreamLanguage.define(_cmake2.cmake);
  },
  cob: function cob() {
    return _language.StreamLanguage.define(_cobol.cobol);
  },
  coffee: function coffee() {
    return _language.StreamLanguage.define(_coffeescript.coffeeScript);
  },
  cpp: function cpp() {
    return (0, _langCpp.cpp)();
  },
  cpy: function cpy() {
    return _language.StreamLanguage.define(_cobol.cobol);
  },
  cql: function cql() {
    return SQL.sql({
      dialect: SQL.Cassandra
    });
  },
  cr: function cr() {
    return _language.StreamLanguage.define(_crystal.crystal);
  },
  cs: function cs() {
    return _language.StreamLanguage.define(_clike.csharp);
  },
  css: function css() {
    return (0, _langCss.css)();
  },
  cts: function cts() {
    return (0, _langJavascript.javascript)({
      typescript: true
    });
  },
  cxx: function cxx() {
    return (0, _langCpp.cpp)();
  },
  cyp: function cyp() {
    return _language.StreamLanguage.define(_cypher2.cypher);
  },
  cypher: function cypher() {
    return _language.StreamLanguage.define(_cypher2.cypher);
  },
  d: function d() {
    return _language.StreamLanguage.define(_d2.d);
  },
  dart: function dart() {
    return _language.StreamLanguage.define(_clike.dart);
  },
  diff: function diff() {
    return _language.StreamLanguage.define(_diff2.diff);
  },
  dtd: function dtd() {
    return _language.StreamLanguage.define(_dtd2.dtd);
  },
  dyalog: function dyalog() {
    return _language.StreamLanguage.define(_apl2.apl);
  },
  dyl: function dyl() {
    return _language.StreamLanguage.define(_dylan2.dylan);
  },
  dylan: function dylan() {
    return _language.StreamLanguage.define(_dylan2.dylan);
  },
  e: function e() {
    return _language.StreamLanguage.define(_eiffel.eiffel);
  },
  ecl: function ecl() {
    return _language.StreamLanguage.define(_ecl2.ecl);
  },
  edn: function edn() {
    return _language.StreamLanguage.define(_clojure.clojure);
  },
  el: function el() {
    return _language.StreamLanguage.define(_commonlisp.commonLisp);
  },
  elm: function elm() {
    return _language.StreamLanguage.define(_elm2.elm);
  },
  erl: function erl() {
    return _language.StreamLanguage.define(_erlang.erlang);
  },
  f: function f() {
    return _language.StreamLanguage.define(_fortran.fortran);
  },
  f77: function f77() {
    return _language.StreamLanguage.define(_fortran.fortran);
  },
  f90: function f90() {
    return _language.StreamLanguage.define(_fortran.fortran);
  },
  f95: function f95() {
    return _language.StreamLanguage.define(_fortran.fortran);
  },
  factor: function factor() {
    return _language.StreamLanguage.define(_factor2.factor);
  },
  feature: function feature() {
    return _language.StreamLanguage.define(_gherkin.gherkin);
  },
  "for": function _for() {
    return _language.StreamLanguage.define(_fortran.fortran);
  },
  forth: function forth() {
    return _language.StreamLanguage.define(_forth2.forth);
  },
  fs: function fs() {
    return _language.StreamLanguage.define(_mllike.fSharp);
  },
  fth: function fth() {
    return _language.StreamLanguage.define(_forth2.forth);
  },
  fun: function fun() {
    return _language.StreamLanguage.define(_mllike.sml);
  },
  go: function go() {
    return (0, _langGo.go)();
  },
  gradle: function gradle() {
    return _language.StreamLanguage.define(_groovy2.groovy);
  },
  groovy: function groovy() {
    return _language.StreamLanguage.define(_groovy2.groovy);
  },
  gss: function gss() {
    return _language.StreamLanguage.define(_css2.gss);
  },
  h: function h() {
    return (0, _langCpp.cpp)();
  },
  'h++': function h() {
    return (0, _langCpp.cpp)();
  },
  handlebars: function handlebars() {
    return (0, _langHtml.html)();
  },
  hbs: function hbs() {
    return (0, _langHtml.html)();
  },
  hh: function hh() {
    return (0, _langCpp.cpp)();
  },
  hpp: function hpp() {
    return (0, _langCpp.cpp)();
  },
  hs: function hs() {
    return _language.StreamLanguage.define(_haskell.haskell);
  },
  htm: function htm() {
    return (0, _langHtml.html)();
  },
  html: function html() {
    return (0, _langHtml.html)();
  },
  hx: function hx() {
    return _language.StreamLanguage.define(_haxe.haxe);
  },
  hxml: function hxml() {
    return _language.StreamLanguage.define(_haxe.hxml);
  },
  hxx: function hxx() {
    return (0, _langCpp.cpp)();
  },
  "in": function _in() {
    return _language.StreamLanguage.define(_properties2.properties);
  },
  ini: function ini() {
    return _language.StreamLanguage.define(_properties2.properties);
  },
  ino: function ino() {
    return (0, _langCpp.cpp)();
  },
  intr: function intr() {
    return _language.StreamLanguage.define(_dylan2.dylan);
  },
  j2: function j2() {
    return (0, _langJinja.jinja)();
  },
  jade: function jade() {
    return _language.StreamLanguage.define(_pug2.pug);
  },
  java: function java() {
    return (0, _langJava.java)();
  },
  jinja: function jinja() {
    return (0, _langJinja.jinja)();
  },
  jinja2: function jinja2() {
    return (0, _langJinja.jinja)();
  },
  jl: function jl() {
    return _language.StreamLanguage.define(_julia.julia);
  },
  js: function js() {
    return (0, _langJavascript.javascript)();
  },
  json: function json() {
    return (0, _langJson.json)();
  },
  jsonld: function jsonld() {
    return _language.StreamLanguage.define(_javascript.jsonld);
  },
  jsx: function jsx() {
    return (0, _langJavascript.javascript)({
      jsx: true
    });
  },
  ksh: function ksh() {
    return _language.StreamLanguage.define(_shell.shell);
  },
  kt: function kt() {
    return _language.StreamLanguage.define(_clike.kotlin);
  },
  kts: function kts() {
    return _language.StreamLanguage.define(_clike.kotlin);
  },
  less: function less() {
    return (0, _langLess.less)();
  },
  liquid: function liquid() {
    return (0, _langLiquid.liquid)();
  },
  lisp: function lisp() {
    return _language.StreamLanguage.define(_commonlisp.commonLisp);
  },
  ls: function ls() {
    return _language.StreamLanguage.define(_livescript.liveScript);
  },
  ltx: function ltx() {
    return _language.StreamLanguage.define(_stex.stex);
  },
  lua: function lua() {
    return _language.StreamLanguage.define(_lua2.lua);
  },
  m: function m() {
    return _language.StreamLanguage.define(_octave.octave);
  },
  map: function map() {
    return (0, _langJson.json)();
  },
  markdown: function markdown() {
    return (0, _langMarkdown.markdown)();
  },
  mbox: function mbox() {
    return _language.StreamLanguage.define(_mbox2.mbox);
  },
  md: function md() {
    return (0, _langMarkdown.markdown)();
  },
  mjs: function mjs() {
    return (0, _langJavascript.javascript)();
  },
  mkd: function mkd() {
    return (0, _langMarkdown.markdown)();
  },
  ml: function ml() {
    return _language.StreamLanguage.define(_mllike.oCaml);
  },
  mli: function mli() {
    return _language.StreamLanguage.define(_mllike.oCaml);
  },
  mll: function mll() {
    return _language.StreamLanguage.define(_mllike.oCaml);
  },
  mly: function mly() {
    return _language.StreamLanguage.define(_mllike.oCaml);
  },
  mm: function mm() {
    return _language.StreamLanguage.define(_clike.objectiveCpp);
  },
  mo: function mo() {
    return _language.StreamLanguage.define(_modelica.modelica);
  },
  mps: function mps() {
    return _language.StreamLanguage.define(_mumps.mumps);
  },
  mrc: function mrc() {
    return _language.StreamLanguage.define(_mirc.mirc);
  },
  msc: function msc() {
    return _language.StreamLanguage.define(_mscgen2.mscgen);
  },
  mscgen: function mscgen() {
    return _language.StreamLanguage.define(_mscgen2.mscgen);
  },
  mscin: function mscin() {
    return _language.StreamLanguage.define(_mscgen2.mscgen);
  },
  msgenny: function msgenny() {
    return _language.StreamLanguage.define(_mscgen2.msgenny);
  },
  mts: function mts() {
    return (0, _langJavascript.javascript)({
      typescript: true
    });
  },
  nb: function nb() {
    return _language.StreamLanguage.define(_mathematica.mathematica);
  },
  nix: function nix() {
    return (0, _codemirrorLangNix.nix)();
  },
  nq: function nq() {
    return _language.StreamLanguage.define(_ntriples.ntriples);
  },
  nsh: function nsh() {
    return _language.StreamLanguage.define(_nsis.nsis);
  },
  nsi: function nsi() {
    return _language.StreamLanguage.define(_nsis.nsis);
  },
  nt: function nt() {
    return _language.StreamLanguage.define(_ntriples.ntriples);
  },
  nut: function nut() {
    return _language.StreamLanguage.define(_clike.squirrel);
  },
  oz: function oz() {
    return _language.StreamLanguage.define(_oz2.oz);
  },
  p: function p() {
    return _language.StreamLanguage.define(_pascal.pascal);
  },
  pas: function pas() {
    return _language.StreamLanguage.define(_pascal.pascal);
  },
  patch: function patch() {
    return _language.StreamLanguage.define(_diff2.diff);
  },
  pgp: function pgp() {
    return _language.StreamLanguage.define(_asciiarmor.asciiArmor);
  },
  php: function php() {
    return (0, _langPhp.php)();
  },
  php3: function php3() {
    return (0, _langPhp.php)();
  },
  php4: function php4() {
    return (0, _langPhp.php)();
  },
  php5: function php5() {
    return (0, _langPhp.php)();
  },
  php7: function php7() {
    return (0, _langPhp.php)();
  },
  phtml: function phtml() {
    return (0, _langPhp.php)();
  },
  pig: function pig() {
    return _language.StreamLanguage.define(_pig2.pig);
  },
  pl: function pl() {
    return _language.StreamLanguage.define(_perl.perl);
  },
  pls: function pls() {
    return SQL.sql({
      dialect: SQL.PLSQL
    });
  },
  pm: function pm() {
    return _language.StreamLanguage.define(_perl.perl);
  },
  pp: function pp() {
    return _language.StreamLanguage.define(_puppet.puppet);
  },
  pro: function pro() {
    return _language.StreamLanguage.define(_idl.idl);
  },
  properties: function properties() {
    return _language.StreamLanguage.define(_properties2.properties);
  },
  proto: function proto() {
    return _language.StreamLanguage.define(_protobuf.protobuf);
  },
  ps1: function ps1() {
    return _language.StreamLanguage.define(_powershell.powerShell);
  },
  psd1: function psd1() {
    return _language.StreamLanguage.define(_powershell.powerShell);
  },
  psm1: function psm1() {
    return _language.StreamLanguage.define(_powershell.powerShell);
  },
  pug: function pug() {
    return _language.StreamLanguage.define(_pug2.pug);
  },
  pxd: function pxd() {
    return _language.StreamLanguage.define(_python2.cython);
  },
  pxi: function pxi() {
    return _language.StreamLanguage.define(_python2.cython);
  },
  py: function py() {
    return (0, _langPython.python)();
  },
  python: function python() {
    return (0, _langPython.python)();
  },
  pyw: function pyw() {
    return (0, _langPython.python)();
  },
  pyx: function pyx() {
    return _language.StreamLanguage.define(_python2.cython);
  },
  q: function q() {
    return _language.StreamLanguage.define(_q2.q);
  },
  r: function r() {
    return _language.StreamLanguage.define(_r2.r);
  },
  R: function R() {
    return _language.StreamLanguage.define(_r2.r);
  },
  rb: function rb() {
    return _language.StreamLanguage.define(_ruby.ruby);
  },
  rq: function rq() {
    return _language.StreamLanguage.define(_sparql2.sparql);
  },
  rs: function rs() {
    return (0, _langRust.rust)();
  },
  s: function s() {
    return _language.StreamLanguage.define(_gas.gas);
  },
  sas: function sas() {
    return _language.StreamLanguage.define(_sas2.sas);
  },
  sass: function sass() {
    return (0, _langSass.sass)({
      indented: true
    });
  },
  scala: function scala() {
    return _language.StreamLanguage.define(_clike.scala);
  },
  scm: function scm() {
    return _language.StreamLanguage.define(_scheme.scheme);
  },
  scss: function scss() {
    return (0, _langSass.sass)();
  },
  sh: function sh() {
    return _language.StreamLanguage.define(_shell.shell);
  },
  sieve: function sieve() {
    return _language.StreamLanguage.define(_sieve2.sieve);
  },
  sig: function sig() {
    return _language.StreamLanguage.define(_mllike.sml);
  },
  siv: function siv() {
    return _language.StreamLanguage.define(_sieve2.sieve);
  },
  smackspec: function smackspec() {
    return _language.StreamLanguage.define(_mllike.sml);
  },
  sml: function sml() {
    return _language.StreamLanguage.define(_mllike.sml);
  },
  solidity: function solidity() {
    return _codemirrorLangSolidity.solidity;
  },
  sparql: function sparql() {
    return _language.StreamLanguage.define(_sparql2.sparql);
  },
  spec: function spec() {
    return _language.StreamLanguage.define(_rpm.rpmSpec);
  },
  sql: function sql() {
    return SQL.sql({
      dialect: SQL.StandardSQL
    });
  },
  ss: function ss() {
    return _language.StreamLanguage.define(_scheme.scheme);
  },
  st: function st() {
    return _language.StreamLanguage.define(_smalltalk.smalltalk);
  },
  styl: function styl() {
    return _language.StreamLanguage.define(_stylus.stylus);
  },
  sv: function sv() {
    return _language.StreamLanguage.define(_verilog.verilog);
  },
  svelte: function svelte() {
    return (0, _codemirrorLangSvelte.svelte)();
  },
  svg: function svg() {
    return (0, _langXml.xml)();
  },
  svh: function svh() {
    return _language.StreamLanguage.define(_verilog.verilog);
  },
  swift: function swift() {
    return _language.StreamLanguage.define(_swift2.swift);
  },
  tcl: function tcl() {
    return _language.StreamLanguage.define(_tcl2.tcl);
  },
  tex: function tex() {
    return _language.StreamLanguage.define(_stex.stex);
  },
  text: function text() {
    return _language.StreamLanguage.define(_stex.stex);
  },
  textile: function textile() {
    return _language.StreamLanguage.define(_textile2.textile);
  },
  toml: function toml() {
    return _language.StreamLanguage.define(_toml2.toml);
  },
  ts: function ts() {
    return (0, _langJavascript.javascript)({
      typescript: true
    });
  },
  tsx: function tsx() {
    return (0, _langJavascript.javascript)({
      jsx: true,
      typescript: true
    });
  },
  ttcn: function ttcn() {
    return _language.StreamLanguage.define(_ttcn2.ttcn);
  },
  ttcn3: function ttcn3() {
    return _language.StreamLanguage.define(_ttcn2.ttcn);
  },
  ttcnpp: function ttcnpp() {
    return _language.StreamLanguage.define(_ttcn2.ttcn);
  },
  ttl: function ttl() {
    return _language.StreamLanguage.define(_turtle.turtle);
  },
  v: function v() {
    return _language.StreamLanguage.define(_verilog.verilog);
  },
  vb: function vb() {
    return _language.StreamLanguage.define(_vb2.vb);
  },
  vbs: function vbs() {
    return _language.StreamLanguage.define(_vbscript.vbScript);
  },
  vhd: function vhd() {
    return _language.StreamLanguage.define(_vhdl2.vhdl);
  },
  vhdl: function vhdl() {
    return _language.StreamLanguage.define(_vhdl2.vhdl);
  },
  vtl: function vtl() {
    return _language.StreamLanguage.define(_velocity.velocity);
  },
  vue: function vue() {
    return (0, _langVue.vue)();
  },
  wast: function wast() {
    return (0, _langWast.wast)();
  },
  wat: function wat() {
    return (0, _langWast.wast)();
  },
  webidl: function webidl() {
    return _language.StreamLanguage.define(_webidl.webIDL);
  },
  wl: function wl() {
    return _language.StreamLanguage.define(_mathematica.mathematica);
  },
  wls: function wls() {
    return _language.StreamLanguage.define(_mathematica.mathematica);
  },
  xml: function xml() {
    return (0, _langXml.xml)();
  },
  xq: function xq() {
    return _language.StreamLanguage.define(_xquery.xQuery);
  },
  xqm: function xqm() {
    return _language.StreamLanguage.define(_xquery.xQuery);
  },
  xquery: function xquery() {
    return _language.StreamLanguage.define(_xquery.xQuery);
  },
  xqy: function xqy() {
    return _language.StreamLanguage.define(_xquery.xQuery);
  },
  xsd: function xsd() {
    return (0, _langXml.xml)();
  },
  xsl: function xsl() {
    return (0, _langXml.xml)();
  },
  xu: function xu() {
    return _language.StreamLanguage.define(_mscgen2.xu);
  },
  xy: function xy() {
    return _language.StreamLanguage.define(_xquery.xQuery);
  },
  yaml: function yaml() {
    return (0, _langYaml.yaml)();
  },
  yml: function yml() {
    return (0, _langYaml.yaml)();
  },
  ys: function ys() {
    return _language.StreamLanguage.define(_yacas.yacas);
  },
  z80: function z80() {
    return _language.StreamLanguage.define(_z2.z80);
  }
};
var langNames = exports.langNames = Object.keys(langs);
function loadLanguage(name) {
  return langs[name] ? langs[name]() : null;
}
